ui/app/adapters/variable.js

```diff
@@ -17,6 +22,65 @@ export default class VariableAdapter extends ApplicationAdapter {
     return this.ajax(`${baseUrl}?cas=${checkAndSetValue}`, 'PUT', { data });
   }
 
+  /**
+   * Query for job templates, both defaults and variables at the nomad/job-templates path.
+   * @returns {Promise<{variables: Variable[], default: Variable[]}>}
+   */
+  async getJobTemplates() {
+    this.populateDefaultJobTemplates();
```

Comment:
`nit`: This should have an `await` to ensure that we populate the store before moving onto the next line of code.